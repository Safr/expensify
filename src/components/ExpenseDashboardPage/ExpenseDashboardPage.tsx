import React, { Component } from 'react';
import { connect } from 'react-redux';
import ExpenseList from '../ExpenseList/ExpenseList';
import ExpensesSummary from '../ExpensesSummary/ExpensesSummary';
import ExpenseListFilters from '../ExpenseListFilters/ExpenseListFilters';
import LoadingPage from '../LoadingPage/LoadingPage';
import StoreState from '../../types';

interface Props {
  loading: boolean;
}

export class ExpenseDashboardPage extends Component<Props, {}> {
  render() {
    return (
     <>
      {this.props.loading ?
         <LoadingPage />
       :
      (<>
        <ExpensesSummary />
        <ExpenseListFilters />
        <ExpenseList />
        </>
      )}
      </>
    );
  }
}

const mapStateToProps = (state: StoreState) => ({
  loading: state.async.loading,
});

export default connect(mapStateToProps)(ExpenseDashboardPage);
