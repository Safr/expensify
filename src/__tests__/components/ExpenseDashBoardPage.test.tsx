import React from 'react';
import { ExpenseDashboardPage } from '../../components/ExpenseDashboardPage/ExpenseDashboardPage';

describe('ExpenseDashboard Component', () => {
  test('should render correctly', () => {
    const wrapper = shallow(<ExpenseDashboardPage />);
    expect(wrapper).toMatchSnapshot();
  });
});
