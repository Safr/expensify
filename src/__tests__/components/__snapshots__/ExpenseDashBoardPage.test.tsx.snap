// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpenseDashboard Component should render correctly 1`] = `
<React.Fragment>
  <React.Fragment>
    <Connect(Component) />
    <Connect(ExpenseListFilters) />
    <Connect(Component) />
  </React.Fragment>
</React.Fragment>
`;
